### ISSUE TITLE
Button to Toggle Activation

### ISSUE THREAD

**Issue by PossiblyRemember** [studio-minus/ppg-bugs-and-requests#1565]
**Feature description**
Like pressing F to activate but instead toggles activation like an activation toggle.

**Feature purpose**
It gets annoying having to use the activation toggle and a propagation wire to keep things activated while you do something else.

**Additional context**
This is the best way I could explain this.

**Comment by PossiblyRemember** [comment]
not really, I meant a feature like the activation toggle block, but able to be activated on the press of a key

**Comment by DindinYT37** [comment]
> not really, I meant a feature like the activation toggle block, but able to be activated on the press of a key

That's just a Key Trigger running into an Activation Block...

**Comment by PossiblyRemember** [comment]
> > yes I get that, just don't see why I'd need to place all of that stuff to toggle something.
> 
> "all of that", mate that's literally two things. Your only reason to add something like an "Key Activation Block" is that you're too literally too lazy to place two things and connect them with a wire.

ye little bit lazy, but that's besides the point. it's a feature that would be nice to have for the ability to do it faster, even if it doesn't take too long

**Comment by DindinYT37** [comment]
> > > yes I get that, just don't see why I'd need to place all of that stuff to toggle something.
> > 
> > 
> > "all of that", mate that's literally two things. Your only reason to add something like an "Key Activation Block" is that you're too literally too lazy to place two things and connect them with a wire.
> 
> ye little bit lazy, but that's besides the point. it's a feature that would be nice to have for the ability to do it faster, even if it doesn't take too long

Not a little bit lazy, very lazy. Not beside the point either, that's the whole point. Do it faster? Copy-paste it then. Doesn't take too long? Why even bother then.

**Comment by PossiblyRemember** [comment]
> > > > yes I get that, just don't see why I'd need to place all of that stuff to toggle something.
> > > 
> > > 
> > > "all of that", mate that's literally two things. Your only reason to add something like an "Key Activation Block" is that you're too literally too lazy to place two things and connect them with a wire.
> > 
> > 
> > ye little bit lazy, but that's besides the point. it's a feature that would be nice to have for the ability to do it faster, even if it doesn't take too long
> 
> Not a little bit lazy, very lazy. Not beside the point either, that's the whole point. Do it faster? Copy-paste it then. Doesn't take too long? Why even bother then.

could you plz not call me lazy I'm trying to be as nice here as I can, it would be a nice feature in my mind it's not really logical to argue about it, zooi will decide

**Comment by DindinYT37** [comment]
> > Not a little bit lazy, very lazy. Not beside the point either, that's the whole point. Do it faster? Copy-paste it then. Doesn't take too long? Why even bother then.
> 
> could you plz not call me lazy I'm trying to be as nice here as I can, it would be a nice feature in my mind it's not really logical to argue about it, zooi will decide

I'm not being rude, I'm saying that you are lazy because you are acting so. If you really wanna talk about something not being logical, then you should know that this request is not logical at all.

The only reason zooi may add this would **probably** just because he could kinda merge the code of both objects and it'd **likely** not be that much work.

**Comment by PossiblyRemember** [comment]
wait did you actually get what you meant because looking over I may not have explained it the best way

**Comment by DindinYT37** [comment]
> wait did you actually get what you meant because looking over I may not have explained it the best way

That sounds very much like an excuse but as far as I have understood, you want a Key Trigger that instead of sending a single signal toggles a continuous signal.

**Comment by PossiblyRemember** [comment]
not exactly, what I meant was you could press a button on your keyboard to toggle something, similar to pressing f but toggle, sorry I made it unclear

**Comment by DindinYT37** [comment]
> not exactly, what I meant was you could press a button on your keyboard to toggle something, similar to pressing f but toggle, sorry I made it unclear. if it were that then it really would be a stupid idea

That's literally the same goddamn thing.

**Comment by DindinYT37** [comment]
> really you don't think this is a good idea?

Well. "if it were that then it really would be a stupid idea" - Literally you, just now.

**Comment by PossiblyRemember** [comment]
ok I don't mean this being toggle
![image](https://user-images.githubusercontent.com/105954143/176781602-e6b9f7d9-36bb-49c2-8bbd-8964c62918f5.png)
that's what I was talking about

**Comment by DindinYT37** [comment]
> ok I don't mean this being toggle
> ![image](https://user-images.githubusercontent.com/105954143/176781602-e6b9f7d9-36bb-49c2-8bbd-8964c62918f5.png)
> that's what I was talking about

Then tf do you mean???

**Comment by PossiblyRemember** [comment]
this
![image](https://user-images.githubusercontent.com/105954143/176782217-4348f9f1-431b-4610-acfe-07068336e733.png)
but it toggles instead of activating once

**Comment by DindinYT37** [comment]
> this
> ![image](https://user-images.githubusercontent.com/105954143/176782217-4348f9f1-431b-4610-acfe-07068336e733.png)
> but it toggles instead of activating once

Finally an explanation that makes sense. Well what key would you assign it to? Or just shift+F? Also how would one know if something was toggled, is there an outline in detailed mode? Can you toggle it per context menu too?

**Comment by PossiblyRemember** [comment]
sorry I was gone. I was thinking something like shift+F. maybe you could see if it was toggled in detail mode, and you'd be able to toggle it in the context menu

**Comment by PossiblyRemember** [comment]
alright, goodnight

**Comment by DindinYT37** [comment]
@PossiblyRemember I suggest hiding this discussion as "resolved".

**Comment by DindinYT37** [comment]
Not what I meant but alright.

**Comment by PossiblyRemember** [comment]
ok how do I make it resolved
